### PR TITLE
PUD-247 - Extend E2E for sentence info fields

### PR DESCRIPTION
**Pull request by jonnywyatt**
Including sentence length for PUD-410

### CODE REVIEW

File: src/test/java/cucumber/steps/NavigationSteps.java [hunk 1]
```diff
@@ -161,12 +161,26 @@ public void submitDateTimeOfRecallRequestReceived(String customer) {
...
-    @When("{word} submits the latest release date and releasing prison details")
+    @When("{word} submits the sentence, offence and release details")
     public void submitLatestReleaseDetails(String customer) {
-        userEnters(customer, LastReleaseDetailsPage.RELEASING_PRISON, "Belmarsh");
-        userEnters(customer, LastReleaseDetailsPage.DAY_FIELD, "03");
-        userEnters(customer, LastReleaseDetailsPage.MONTH_FIELD, "08");
-        userEnters(customer, LastReleaseDetailsPage.YEAR_FIELD, "2020");
+        userEnters(customer, LastReleaseDetailsPage.dateDayInput("sentenceDate"), "03");
+        userEnters(customer, LastReleaseDetailsPage.dateMonthInput("sentenceDate"), "08");
+        userEnters(customer, LastReleaseDetailsPage.dateYearInput("sentenceDate"), "2020");
```
Comment (philsegal) [inline src/test/java/cucumber/steps/NavigationSteps.java:168]: We definitely need to abstract a date fieldset concept.

File: src/test/java/cucumber/pages/PpudPage.java [hunk 2]
```diff
@@ -8,15 +8,25 @@
...
         return Target.the(value + " matches").locatedBy("[value='" + value + "']");
     }
 
-    protected static Target getTargetByName(String name) {
+    public static Target getTargetByName(String name) {
         return Target.the(name + " matches").locatedBy("[name='" + name + "']");
     }
+
+    public static Target dateYearInput(String fieldsetName) {
+        return Target.the(fieldsetName + " matches").locatedBy("[name='" + fieldsetName + "Year']");
```
Comment (philsegal) [inline src/test/java/cucumber/pages/PpudPage.java:24]: should use `By.name(fieldSetName + "Year")` etc

File: src/test/java/cucumber/pages/PpudPage.java [hunk 3]
```diff
@@ -8,15 +8,25 @@
...
         return Target.the(dataQa + " matches").locatedBy("[data-qa='" + dataQa + "']");
     }
 
     protected static Target getTargetByInputValue(String value) {
         return Target.the(value + " matches").locatedBy("[value='" + value + "']");
     }
 
-    protected static Target getTargetByName(String name) {
+    public static Target getTargetByName(String name) {
         return Target.the(name + " matches").locatedBy("[name='" + name + "']");
```
Comment (philsegal) [inline src/test/java/cucumber/pages/PpudPage.java:20]: use By.name (whilst we're here)

File: src/test/java/cucumber/pages/PpudPage.java (hunk 2 above)
Comment (jonnywyatt) [reply src/test/java/cucumber/pages/PpudPage.java:24]: Can you post the full syntax?
`return Target.the(fieldsetName + " matches").By.name(fieldsetName + "Year");`
throws an error
I'd search the serenity manual but it has no search function 🥇

File: src/test/java/cucumber/pages/PpudPage.java (hunk 3 above)
Comment (jonnywyatt) [reply src/test/java/cucumber/pages/PpudPage.java]: done

File: src/test/java/cucumber/pages/PpudPage.java (hunk 2 above)
Comment (jonnywyatt) [reply src/test/java/cucumber/pages/PpudPage.java]: done

File: src/test/java/cucumber/steps/NavigationSteps.java (hunk 1 above)
Comment (jonnywyatt) [reply src/test/java/cucumber/steps/NavigationSteps.java:168]: yep that was my first thought but my java is so bad I just took a step towards it

**Review by philsegal — APPROVED** [review]

**Review by philsegal — APPROVED** [review]

**Pull request merged**